plugins/modules/inventory_intent.py

```diff
@@ -1992,13 +2033,114 @@ def get_device_ids(self, device_ips):
             except Exception as e:
                 error_message = "Error while fetching device '{0}' from Cisco Catalyst Center: {1}".format(device_ip, str(e))
                 self.log(error_message, "ERROR")
-                raise Exception(error_message)
 
         return device_ids
 
+    def get_device_ips_from_hostname(self, hostname_list):
+        """
+        Get the list of unique device IDs for list of specified hostnames of devices in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            hostname_list (list): The hostnames of devices for which you want to retrieve the device IDs.
+        Returns:
+            list: The list of unique device IDs for the specified devices hostname list.
+        Description:
+            Queries Cisco Catalyst Center to retrieve the unique device ID associated with a device having the specified
+            list of hostnames. If the device is not found in Cisco Catalyst Center, tthen print the log message with error severity.
+        """
+
+        device_ips = []
+        for hostname in hostname_list:
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"hostname": hostname}
+                )
+                if response:
+                    self.log("Received API response from 'get_device_list': {0}".format(str(response)), "DEBUG")
+                    response = response.get("response")
+                    if not response:
+                        continue
+                    device_ip = response[0]["managementIpAddress"]
+                    device_ips.append(device_ip)
+            except Exception as e:
+                error_message = "Error while fetching device from Cisco Catalyst Center - {0}".format(str(e))
+                self.log(error_message, "ERROR")
+
+        return device_ips
+
+    def get_device_ips_from_serial_number(self, serial_number_list):
+        """
+        Get the list of unique device IDs for list of specified serial number of devices in Cisco Catalyst Center.
```

Comment:
device IPs ... Also, it would be more meaningful if we say "for a specified list of serial numbers" instead of 
Example:
`  Get the list of unique device IPs for a specified list of serial numbers in Cisco Catalyst Center.
`

Reply:
added.